backend/src/api/classes/classes.controller.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+  Query,
+} from "@nestjs/common";
+import {
+  ApiCreatedResponse,
+  ApiForbiddenResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiQuery,
+  ApiTags,
+} from "@nestjs/swagger";
+import { plainToInstance } from "class-transformer";
+import {
+  CreateClassDto,
+  ExistingClassDto,
+  UpdateClassDto,
+  DeletedClassDto,
+  ClassId,
+  ExistingClassTeacherDto,
+  ExistingClassExtendedDto,
+} from "./dto";
+import { ClassesService } from "./classes.service";
+
+@Controller("classes")
+@ApiTags("classes")
+export class ClassesController {
+  constructor(private readonly classesService: ClassesService) {}
+
+  @Post()
+  @ApiCreatedResponse({ type: ExistingClassDto })
+  @ApiForbiddenResponse()
+  async create(
+    @Body() createClassDto: CreateClassDto,
+  ): Promise<ExistingClassDto> {
+    const klass = await this.classesService.create(createClassDto);
+    return plainToInstance(ExistingClassDto, klass);
+  }
+
+  @Get()
+  @ApiQuery({ name: "teacherId", required: false, type: Number })
+  @ApiOkResponse({ type: ExistingClassTeacherDto, isArray: true })
+  async findAll(
+    @Query("teacherId", new ParseIntPipe({ optional: true }))
+    teacherId?: number,
+  ): Promise<ExistingClassTeacherDto[]> {
+    // TODO: add pagination support
+
+    const classes = await this.classesService.listClassesWithTeacher({
+      // TODO: Implement this properly when auth is available
+      // TODO: Only allow teachers to see their own classes
```

Comment:
and once we do that, do we still include the teacher in the response?

Reply:
I'd say so. Generic endpoint, same handling on the frontend, just an authz limitation.

Reply:
yes, makes sense!